File: app/src/main/java/com/karla/learningverbs/view/SplashActivity.java
package com.karla.learningverbs.view;

import androidx.annotation.NonNull;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;

import android.animation.Animator;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;

import com.karla.learningverbs.view.home.HomeActivity;
import com.karla.learningverbs.view.profile.CreateAccountActivity;
import com.karla.learningverbs.viewmodel.SplashViewModel;
import com.karla.learningverbs.utils.base.BaseActivity;

public class SplashActivity extends BaseActivity<ActivitySplashBinding, SplashViewModel> {

    @Override
    protected SplashViewModel createViewModel() {
        return new ViewModelProvider(this).get(SplashViewModel.class);
    }

    @NonNull
    @Override
    protected ActivitySplashBinding createViewBinding(LayoutInflater layoutInflater) {
        return ActivitySplashBinding.inflate(layoutInflater);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);


        viewModel.getIsLogin().observe(SplashActivity.this, new Observer<Boolean>() {
            @Override
            public void onChanged(Boolean isLogin) {
                if (isLogin) {
                    startActivity(new Intent(SplashActivity.this, HomeActivity.class));
                } else {
                    startActivity(new Intent(SplashActivity.this, CreateAccountActivity.class));
                }
                finish();
            }
        });


        binding.splashLottie.addAnimatorListener(new Animator.AnimatorListener() {
            @Override
            public void onAnimationStart(@NonNull Animator animator) {


            }

            @Override
            public void onAnimationEnd(@NonNull Animator animator) {
                viewModel.isLogin();

            }

            @Override
            public void onAnimationCancel(@NonNull Animator animator) {

            }

            @Override
            public void onAnimationRepeat(@NonNull Animator animator) {

            }
        });


    }

}

File: app/src/main/java/com/karla/learningverbs/view/SplashActivity.kt
package com.karla.learningverbs.view

import android.animation.Animator
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.karla.learningverbs.databinding.ActivitySplashBinding
import com.karla.learningverbs.utils.base.BaseActivity
import com.karla.learningverbs.view.home.HomeActivity
import com.karla.learningverbs.view.profile.CreateAccountActivity
import com.karla.learningverbs.viewmodel.SplashViewModel

class SplashActivity : BaseActivity<ActivitySplashBinding, SplashViewModel>(){

    override fun createViewModel(): SplashViewModel {
        return ViewModelProvider(this).get(SplashViewModel::class.java)
    }

    override fun createViewBinding(layoutInflater: LayoutInflater): ActivitySplashBinding {
        return ActivitySplashBinding.inflate(layoutInflater)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        viewModel.getIsLogin().observe(this, Observer { isLogin ->
            if (isLogin) {
                startActivity(Intent(this@SplashActivity, HomeActivity::class.java))
            } else {
                startActivity(Intent(this@SplashActivity, CreateAccountActivity::class.java))
            }
            finish()
        })

        binding.splashLottie.addAnimatorListener(object : Animator.AnimatorListener {
            override fun onAnimationStart(animator: Animator) {
                // Implementación del método si es necesario
            }

            override fun onAnimationEnd(animator: Animator) {
                viewModel.isLogin()
            }

            override fun onAnimationCancel(animator: Animator) {
                // Implementación del método si es necesario
            }

            override fun onAnimationRepeat(animator: Animator) {
                // Implementación del método si es necesario
            }
        })
    }
}